components/apps_form/apps_form_field/index.ts

```diff
@@ -24,7 +23,7 @@ function mapStateToProps(state: GlobalState) {
 }
 type Actions = {
     autocompleteChannels: (term: string, success: (channels: Channel[]) => void, error: () => void) => (dispatch: any, getState: any) => Promise<void>;
-    autocompleteUsers: (search: string) => Promise<UserProfile[]>;
+    autocompleteUsers: (search: string) => (doDispatch: DispatchFunc) => Promise<any>;
```

Comment:
In order to avoid repetition, you can actually delete `Actions` type declaration here. 
You can export `Props` type from `apps_form_field.tsx` and import here. Then you can use it like `Props['actions']`